README.md

```diff
@@ -30,7 +30,7 @@ If you use knnSampler, please cite the original [publication](https://arxiv.org/
 
 ### Prerequisites
 
-This project requires Python 3.8+ and [Poetry](https://python-poetry.org/).
+This project requires Python 3.12+ and [Poetry](https://python-poetry.org/).
```

Comment:
```suggestion
This project requires Python 3.12+ and [Poetry](https://python-poetry.org/) 2+.
```

And I do not find any ways to enforce that version in the `pyproject.yaml` file without breaking poetry.